onnxruntime/core/framework/sequential_executor.cc

```diff
@@ -188,7 +368,8 @@ Status SequentialExecutor::Execute(const SessionState& session_state, const std:
       profile::Color::Black);
 #endif
 
-  for (const auto& node_exec_plan : exec_plan_vec) {
+  for (size_t program_counter = program_counter_start; program_counter <= program_counter_end; program_counter += 1) {
```

Comment:
><= [](start = 71, length = 2)

I saw there is some computation for setting the program_counter_end above. 
Maybe it's because of the <=, inclusive end index here. 

You can try and see if it's better to make it an exclusive end index. it's also more intuituve, since we are more  farmilar with the array indexing pattern.